src/site/content/en/blog/codelab-sign-up-form-best-practices/index.md

```diff
@@ -0,0 +1,281 @@
+
+{% Aside %}
+[Browser Default Styles](https://browserdefaultstyles.com) lists default CSS for HTML elements.
+{% endAside %}
+
+### Test your form
+
+You could acquire a lot of hardware and set up a 
+[device lab](https://www.smashingmagazine.com/2016/11/worlds-best-open-device-labs/), but there are 
+cheaper and simpler ways to try out your form on a range of browsers, platforms and devices:
+
+* [Use Chrome DevTools Device Mode](https://developers.google.com/web/tools/chrome-devtools/device-mode) 
+to simulate mobile devices. 
+* [Send the URL from your computer to your phone](https://support.google.com/chrome/answer/9430554).
+* Use a service such as [BrowserStack](https://www.browserstack.com/open-source) to test on a range 
+of devices and browsers. 
+* Try out the form using a screenreader tool such as the [ChromeVox](https://chrome.google.com/webstore/detail/chromevox-classic-extensi/kgejglhpjiefppelpmljglcjbhoiplfn?hl=en) Chrome extension.
+
+{% Aside 'warning' %}
+Testing your site on a range of devices and browsers is especially important for forms, because 
+[small problems can cause high bounce rates](https://baymard.com/checkout-usability) and cause users 
+to give up on creating an account or completing a purchase.
+{% endAside %}
+
+Click **View App** to preview your sign-up form. 
+
+* Try out your form on different devices using Chrome DevTools Device Mode.
+* Now open the form on a phone or other real devices. What differences do you see?
+
+## Step 2: Add CSS to make the form work better
+
+Click **View Source** to return to your source code.
+
+There's nothing wrong with the HTML so far, but you need to make sure your form works well for a 
+range of users on mobile and desktop.
+
+In this step you'll add CSS to make the form easier to use.
+
+Copy and paste all the following CSS into `css/main.css` file:
+
+{% Glitch {
+  id: 'sign-up-form-codelab-2',
+  path: 'css/main.css',
+  height: 400,
+  previewSize: 0
+} %}
+
+Click **View App** to see your styled sign-up form. Then click **View Source** to return to 
+`css/main.css`.
+
+* Does this CSS work for a variety of browsers and screen sizes?
+
+* Try adjusting `padding`, `margin`, and `font-size` to suit your test devices.
+
+* The CSS is mobile-first. [Media queries](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Media_queries) 
+are used to apply CSS rules for viewports that are at least `400px` wide, and then again for 
+viewports that are at least `500px` wide. Are these [breakpoints](https://www.browserstack.com/guide/responsive-design-breakpoints) 
+adequate? How should you choose breakpoints for forms?
+
+{% Aside %}
+**Viewport** in this context means the display area available for your page. Mobile phones have a 
+smaller viewport than laptops, and a small browser window on a desktop monitor has a smaller 
+viewport than a maximized browser window.
+{% endAside %}
+
+
+## Step 3: Add attributes to enable built-in browser features
```

Comment:
I think a more specific subheading would work better here (like "Help password managers securely suggest and store passwords" in the main post).

Reply:
I've changed this to "Add attributes to help users enter data". Does that work OK?